include/dlaf/inverse/cholesky/impl.h

```diff
@@ -0,0 +1,540 @@
+
+  pika::execution::experimental::start_detached(
+      dlaf::internal::whenAllLift(blas::Uplo::Upper, std::forward<MatrixTileSender>(matrix_tile),
+                                  std::forward<WsTileSender>(ws_tile)) |
+      tile::lauum_workspace(dlaf::internal::Policy<backend>(priority, thread_stacksize::nostack)));
+}
+
+template <Backend backend, class KKTileSender, class MatrixTileSender>
+void trmm_col_panel_tile(pika::execution::thread_priority priority, KKTileSender&& kk_tile,
+                         MatrixTileSender&& matrix_tile) {
+  using ElementType = dlaf::internal::SenderElementType<KKTileSender>;
+  using pika::execution::thread_stacksize;
+
+  pika::execution::experimental::start_detached(
+      dlaf::internal::whenAllLift(blas::Side::Right, blas::Uplo::Upper, blas::Op::ConjTrans,
+                                  blas::Diag::NonUnit, ElementType(1.0),
+                                  std::forward<KKTileSender>(kk_tile),
+                                  std::forward<MatrixTileSender>(matrix_tile)) |
+      tile::trmm(dlaf::internal::Policy<backend>(priority, thread_stacksize::nostack)));
+}
+
+template <Backend backend, class PanelTileSender, class MatrixTileSender>
+void herk_matrix_tile(pika::execution::thread_priority priority, PanelTileSender&& panel_tile,
+                      MatrixTileSender&& matrix_tile) {
+  using BaseElementType = BaseType<dlaf::internal::SenderElementType<MatrixTileSender>>;
+  using pika::execution::thread_stacksize;
+
+  pika::execution::experimental::start_detached(
+      dlaf::internal::whenAllLift(blas::Uplo::Upper, blas::Op::NoTrans, BaseElementType(1.0),
+                                  std::forward<PanelTileSender>(panel_tile), BaseElementType(1.0),
+                                  std::forward<MatrixTileSender>(matrix_tile)) |
+      tile::herk(dlaf::internal::Policy<backend>(priority, thread_stacksize::nostack)));
+}
+
+template <Backend backend, class RowPanelTileSender, class ColPanelTileSender, class MatrixTileSender>
+void gemm_matrix_tile(pika::execution::thread_priority priority, RowPanelTileSender&& row_tile,
+                      ColPanelTileSender&& col_tile, MatrixTileSender&& matrix_tile) {
+  using ElementType = dlaf::internal::SenderElementType<MatrixTileSender>;
+  using pika::execution::thread_stacksize;
+
+  pika::execution::experimental::start_detached(
+      dlaf::internal::whenAllLift(blas::Op::NoTrans, blas::Op::ConjTrans, ElementType(1.0),
+                                  std::forward<RowPanelTileSender>(row_tile),
+                                  std::forward<ColPanelTileSender>(col_tile), ElementType(1.0),
+                                  std::forward<MatrixTileSender>(matrix_tile)) |
+      tile::gemm(dlaf::internal::Policy<backend>(priority, thread_stacksize::nostack)));
+}
+}
+
+template <Device D>
+constexpr bool replace_lauum() {
+  if constexpr (D == Device::CPU)
+    return 0;
+  return 1;
+}
```

Comment:
```suggestion
template <Device D>
inline constexpr bool replace_lauum = (D == Device::CPU) ? 0 : 1;
```
?

Reply:
I prefer the current function as it is more flexible and more readable in case of complex logic with ifdefs.